tools/msbuild-fuzzer/Program.cs

```diff
@@ -0,0 +1,311 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using Xamarin.ProjectTools;
+
+namespace MSBuild.Fuzzer
+{
+	class Program
+	{
+		static readonly Random random = new Random ();
+		static readonly List<string> installedPackages = new List<string> ();
+		static ProjectBuilder builder;
+		static XamarinFormsAndroidApplicationProject application;
+		static bool needsInstall = true;
+		static string directory;
+
+		static void Main ()
+		{
+			var temp = Path.Combine (Path.GetDirectoryName (typeof (Program).Assembly.Location));
+			using (builder = new ProjectBuilder (Path.Combine ("temp", "Fuzzer")) {
+				AutomaticNuGetRestore = false,
+				CleanupAfterSuccessfulBuild = false,
+				CleanupOnDispose = true,
+				Root = Xamarin.Android.Build.Paths.TestOutputDirectory,
+			}) {
+				directory = Path.GetFullPath (Path.Combine (builder.Root, builder.ProjectDirectory));
+				if (Directory.Exists (directory))
+					Directory.Delete (directory, recursive: true);
+
+				application = new XamarinFormsAndroidApplicationProject ();
+				application.AndroidManifest = application.AndroidManifest.Replace ("<uses-sdk />", "<uses-sdk android:targetSdkVersion=\"28\" />");
+				application.MainActivity = application.DefaultMainActivity.Replace ("//${AFTER_ONCREATE}", "Android.Util.Log.Debug (\"FUZZER\", \"App started!\");");
+				var abis = new string [] { "armeabi-v7a", "arm64-v8a", "x86" };
+				application.SetProperty (KnownProperties.AndroidSupportedAbis, string.Join (";", abis));
+
+				if (!NuGetRestore ()) {
+					Console.WriteLine ("Initial NuGet restore failed!");
+					return;
+				}
+
+				Func<bool> [] operations = {
+					NuGetRestore,
+					Build,
+					DesignTimeBuild,
+					DesignerBuild,
+					Clean,
+					Install,
+					Uninstall,
+					Run,
+					TouchRandomFile,
+					ChangePackageName,
+					AddClass,
+					RemoveClass,
+					RenameClass,
+					AddResource,
+					RemoveResource,
+					RenameResource,
```

Comment:
Feel free to give other ideas for steps to add here

Reply:
do we want to intermingle operations like the DesignerBuild or a DesignTimeBuild and random points to see what happens if a user say edits and file and the DesignTimeBuild happens before a normal one.

Reply:
These run at random:
```csharp
while (true) {
	var operation = operations [random.Next (operations.Length)];
	if (!operation ()) {
		break;
	}
}
```
This list won't be the order they run.